src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -247,6 +257,25 @@ private static IEnumerable<PackageVulnerabilityMetadata> MergeVulnerabilityMetad
             return vulnerabilityMetadatas.FirstOrDefault(v => v != null && v.Any());
         }
 
+        private static string MergePackagePath(IEnumerable<IPackageSearchMetadata> packages)
+        {
+            var packagePaths = packages
+                .Select(packageMetadata =>
+                {
+                    if (packageMetadata is LocalPackageSearchMetadata localPackageSearchMetadata)
+                    {
+                        return localPackageSearchMetadata.PackagePath;
+                    }
+                    if (packageMetadata is PackageSearchMetadataBuilder.ClonedPackageSearchMetadata clonedPackage)
+                    {
+                        return clonedPackage.PackagePath;
+                    }
+                    return string.Empty;
+                })
+                .Where(packagePath => !string.IsNullOrWhiteSpace(packagePath));
```

Comment:
The `where` can be put before the `select`.

Reply:
I changed the linq query completely, I felt like the select was a little confusing. Lemme know what you think